client/ayon_nuke/api/plugin.py

```
@@ -438,16 +440,24 @@ def _get_render_target_enum(self):
             "local": "Local machine rendering",
             "frames": "Use existing frames"
         }
-        if "farm_rendering" in self.instance_attributes:
-            rendering_targets.update({
-                "frames_farm": "Use existing frames - farm",
-                "farm": "Farm rendering",
-            })
+
+        farm_rendering_targets = {
+            "frames_farm": "Use existing frames - farm",
+            "farm": "Farm rendering",
+        }
+
+        if (
+            "farm_rendering" in self.instance_attributes
+            or self.render_target in farm_rendering_targets
+        ):
+            rendering_targets.update(farm_rendering_targets)
```

I don't think we should add farm options just because you have farm value set in settings as default -> it's hiding wrong settings.
```suggestion
        if "farm_rendering" in self.instance_attributes:
            rendering_targets.update(farm_rendering_targets)
```